plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java

```diff
@@ -86,10 +97,15 @@ public CheckpointWriter(TypeManager typeManager, CheckpointSchemaManager checkpo
 
     public void write(ConnectorSession session, CheckpointEntries entries, Path targetPath)
     {
+        Map<String, String> configuration = entries.getMetadataEntry().getConfiguration();
+        boolean writeStatsAsJson = Boolean.parseBoolean(configuration.getOrDefault(DELTA_CHECKPOINT_WRITE_STATS_AS_JSON_PROPERTY, "true"));
```

Comment:
https://docs.databricks.com/delta/optimizations/file-mgmt.html#trade-offs-with-statistics-in-checkpoints

> Since writing statistics in a checkpoint has a cost (usually < a minute even for large tables), there is a tradeoff between the time taken to write a checkpoint and compatibility with Databricks Runtime 7.2 and below. If you are able to upgrade all of your workloads to Databricks Runtime 7.3 LTS or above you can reduce the cost of writing a checkpoint by disabling the legacy JSON statistics.